### PR TITLE
feat: add tx confirm func and create virtual group retry

### PR DESCRIPTION
**Pull request by will-2012**
### Description

Add tx confirm func and create virtual group retry.

### Rationale

Send tx and query tx hash, in order to ensure the tx executed by chain.

### Example

N/A.

### Changes

Notable changes: 
* signer/chain client.

### CODE REVIEW

File: base/gnfd/gnfd_service.go [hunk 1]
```diff
@@ -783,3 +787,70 @@ func (g *Gnfd) VerifyPutObjectPermission(ctx context.Context, account, bucket, o
 	}
 	return false, err
 }
+
+// ConfirmTransaction is used to confirm whether the transaction is on the chain.
+func (g *Gnfd) ConfirmTransaction(ctx context.Context, txHash string) (*sdk.TxResponse, error) {
```
Comment (Fynnss) [inline base/gnfd/gnfd_service.go:792]: It should have a timeout for retry.

Comment (will-2012) [reply base/gnfd/gnfd_service.go:792]: Currently, only retry once and there is a timeout in waitForNextBlock.

**Review by Fynnss — APPROVED** [review]

File: base/gnfd/gnfd_service.go (hunk 1 above)
Comment (Fynnss) [reply base/gnfd/gnfd_service.go:792]: Why only retry once?

Comment (will-2012) [reply base/gnfd/gnfd_service.go:792]: Retrying once basically guarantees whether it is in the current block or the next block.
what is the suggestion here.

Comment (Fynnss) [reply base/gnfd/gnfd_service.go:792]: At least 3 blocks.

**Review by sysvm — APPROVED** [review]
LGTM

File: base/gnfd/gnfd_service.go (hunk 1 above)
Comment (will-2012) [reply base/gnfd/gnfd_service.go:797]: Fixed.

**Pull request merged**